templates/definition/meter/sofarsolar-g3.yaml

```diff
@@ -0,0 +1,135 @@
+template: sofarsolar-g3
+products:
+  - brand: SofarSolar
+    description:
+      generic: HYD 5…20K-3PH
+  - brand: SofarSolar
+    description:
+      generic: HYD 3…6K-EP
+  - brand: SofarSolar
+    description:
+      generic: SOFAR 80…136KTL
+  - brand: SofarSolar
+    description:
+      generic: SOFAR 5…24KTL-G3
+params:
+  - name: usage
+    choice: ["grid", "pv", "battery"]
+  - name: modbus
+    choice: ["rs485", "tcpip"]
+    baudrate: 9600
+    comset: 8N1
+    port: 502
+    id: 1
+  - name: capacity
+    advanced: true
+render: |
+  type: custom
+  {{- if eq .usage "grid" }}
+  power:
+    source: modbus
+    {{- include "modbus" . | indent 2 }}
+    register:
+      address: 0x0488 # ActivePower_PCC_Total
+      type: holding
+      decode: int16
+    scale: -10
+  currents:
+  - source: modbus
+    {{- include "modbus" . | indent 2 }}
+    register:
+      address: 0x0492	# Current_PCC_R
+      type: holding
+      decode: uint16
+    scale: 0.01
+  - source: modbus
+    {{- include "modbus" . | indent 2 }}
+    register:
+      address: 0x049D	# Current_PCC_S
```

Comment:
@premultiply do we have HEX parsing?

Reply:
Works here with the current EVCC version exactly like that.

Reply:
Sachen gibts :)